test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/DefaultPageApplicationModelProviderTest.cs

```diff
@@ -306,6 +305,53 @@ public void OnProvidersExecuting_DiscoversPropertiesFromModel()
         }
 
         [Fact]
+        public void OnProvidersExecuting_DiscoversBindingInfoFromHandler()
+        {
+            // Arrange
+            var provider = new DefaultPageApplicationModelProvider();
+            var typeInfo = typeof(PageWithBindPropertyModel).GetTypeInfo();
+            var modelType = typeof(ModelWithBindProperty);
+            var descriptor = new PageActionDescriptor();
+            var context = new PageApplicationModelProviderContext(descriptor, typeInfo);
+
+            // Act
+            provider.OnProvidersExecuting(context);
+
+            // Assert
+            Assert.NotNull(context.PageApplicationModel);
+            Assert.Collection(
+                context.PageApplicationModel.HandlerProperties.OrderBy(p => p.PropertyName),
+                property =>
+                {
+                    Assert.Equal(nameof(ModelWithBindProperty.Property1), property.PropertyName);
+                    Assert.NotNull(property.BindingInfo);
+                },
+                property =>
+                {
+                    Assert.Equal(nameof(ModelWithBindProperty.Property2), property.PropertyName);
+                    Assert.NotNull(property.BindingInfo);
+                    Assert.Equal(BindingSource.Path, property.BindingInfo.BindingSource);
+                });
+        }
+
+        private class PageWithBindPropertyModel : PageBase
+        {
+            public ModelWithBindProperty Model => null;
+
+            public override Task ExecuteAsync() => null;
+        }
+
+        [BindProperty]
+        [PageModel]
+        private class ModelWithBindProperty
```

Comment:
Can you add a test where this inherits from `PageModel`. It's **critical** that we don't automatically make all of the base-class properties bindable by default